specs/020-conversation-consolidation/spec.md

```diff
@@ -0,0 +1,74 @@
+**Status**: Implemented (Consolidated)
+
+## Overview
+
+This specification consolidates a major refactoring of the conversation domain and communication infrastructure. It unifies conversation management under a single platform-owned set, introduces a flexible membership model via a pivot table, upgrades the Matrix Adapter to use the new Go-based library with a unified "Actor" pattern, and establishes proper authorization inheritance from the Platform.
+
+## User Scenarios & Testing
+
+### User Story 1 - Platform-Owned Conversations (Priority: P0)
+
+**From Spec 022 & 024**: All conversations must belong to a single `ConversationsSet` owned by the `Platform`. This replaces the legacy model of per-user conversation sets. The `ConversationsSet` must inherit authorization policies from the `Platform`, ensuring global admins have appropriate access.
+
+**Acceptance Scenarios**:
+1.  **Given** a platform instance, **When** the system starts, **Then** it ensures a `ConversationsSet` exists and is linked to the `Platform`.
+2.  **Given** a new conversation, **When** created, **Then** it is assigned to the platform's `ConversationsSet`.
+3.  **Given** a global admin, **When** accessing the `ConversationsSet`, **Then** access is granted via inheritance from the Platform policy.
+
+### User Story 2 - Flexible Membership Model (Priority: P0)
+
+**From Spec 022**: Conversation membership is tracked via a `ConversationMembership` pivot table (linking `Conversation` and `Agent`), replacing rigid `userID` and `virtualContributorID` columns. This supports future multi-party conversations and normalizes the data model.
+
+**Acceptance Scenarios**:
+1.  **Given** a conversation between User A and User B, **When** created, **Then** two rows are added to `conversation_membership` linking the conversation to their respective Agent IDs.
+2.  **Given** a query for conversation members, **When** executed, **Then** it retrieves agents via the pivot table.
+3.  **Given** legacy columns (`userID`, `virtualContributorID`), **When** the migration runs, **Then** these columns are dropped.
+
+### User Story 3 - Matrix Adapter Upgrade (Priority: P1)
+
+**From Spec 021**: The `CommunicationAdapter` is refactored to use `@alkem-io/matrix-adapter-go-lib`. It adopts the "Unified Actor Pattern," using `Agent.id` as the `actorId` for all communication operations, eliminating the distinction between Users and Virtual Contributors at the adapter level.
+
+**Acceptance Scenarios**:
+1.  **Given** a message send request, **When** processed, **Then** the adapter uses `agent.id` as the `sender_actor_id`.
+2.  **Given** a room creation, **When** processed, **Then** it uses `AlkemioRoomID` (UUID) as the identifier.
+3.  **Given** a space creation, **When** processed, **Then** it uses `AlkemioContextID` (Authorization ID) as the identifier.
```

Comment:
Might need a discussion

Reply:
@techsmyth we never touched that. But spaces are not in use at the moment. And the rest is what it is